Packs/Salesforce/Integrations/Salesforce/Salesforce.py

```diff
@@ -0,0 +1,1037 @@
+            comments = self.sendRequestInSession("GET", f"sobjects/Case/{cases[0]['Id']}/CaseComments")
+        elif oid:
+            comments = self.sendRequestInSession("GET", f"sobjects/Case/{oid}/CaseComments")
+
+        return comments
+
+    def getUser(self, oid, caseNumber):
+
+        if caseNumber:
+            condition = f"CaseNumber='{caseNumber}'"
+            cases = self.queryObjects(['Id', 'CaseNumber', 'OwnerId'], 'Case', condition).get("records", [])
+            if len(cases) == 1:
+                # retrieve object type based on OwnerId
+                obj_type = self.identifyObjectType(cases[0].get('OwnerId'))
+                users = self.getObject(f"{obj_type}/{cases[0].get('OwnerId')}")
+        elif oid:
+            users = self.getObject(f"User/{oid}")
+
+        return users
+
+    # Add the capability to post comment in specific case
+    def postCaseComment(self, public, oid, caseNumber, text):
+
+        # retrieve oid based on case number
+
+        if caseNumber:
+            condition = f"CaseNumber='{caseNumber}'"
+            cases = self.queryObjects(['Id', 'CaseNumber', 'OwnerId'], 'Case', condition).get("records", [])
+            if len(cases) == 1:
+                oid = cases[0].get('Id')
+            else:
+                return_error('Invalid Case Number Provided.')
+
+        data = {
+            "CommentBody": text,
+            "ParentId": oid,
+            "IsPublished": public
+        }
+
+        response = self.sendRequestInSession('POST', 'sobjects/CaseComment', data)
+
+        return response
+
+    def pushComment(self, oid, text, linkUrl):
+        data = {
+            "body": {
+                "messageSegments": [{
+                    "type": 'Text',
+                    "text": text
+                }],
+            },
+            "feedElementType": 'FeedItem',
+            "subjectId": oid
+        }
+
+        if linkUrl:
+            data['body']['messageSegments'].append({
+                "type": 'Link',
+                "url": linkUrl})
+
+        response = self.sendRequestInSession('POST', 'chatter/feed-elements', data)
+
+        return response
+
+    def pushCommentThread(self, id, text):
+        data = {
+            "body": {
+                "messageSegments": [{
+                    "type": 'Text',
+                    "text": text
+                }]
+            }
+        }
+
+        response = self.sendRequestInSession('POST', f'chatter/feed-elements/{id}/capabilities/comments/items', data)
+
+        return response
+
+
+''' HELPER FUNCTIONS '''
+
+
+def getUserNames(client):
+    res = client.queryObjects(['Id', 'Name'], 'User', None)
+    users = {}
+    for item in res.get('records'):
+        users[item.get('Id')] = item.get('Name')
+
+    return users
+
+
+def searchToEntry(client, searchRecords):
+    if len(searchRecords) == 0:
+        return_error('No records matched the search.')
```

Comment:
i suggest to raise exceptions and not to use here ```return_error```. Catch all the exceptions in main and just use there ```return_error```